### ISSUE TITLE
Flow event and action

### ISSUE THREAD

**Issue by blizzz** [nextcloud/spreed#2414]
We would like to equip Talk for Nc 18 with following events and actions:

## Events

* [ ] call has finished (provide conversation name, start and end date, participants, conversation message in this time frame)

## Actions (Operations)

* [x] write a message into a Talk conversation
* [ ] create a Talk notification (so probably like ^ but one on one)

The actions should be independent of the triggering event, nevertheless should support/consider some context (e.g. the event itself like file modified, file path/name for Files, or also tag when in combination with a tagging event).

An Action (Operation in tech terms) implementation to look at could be https://github.com/nextcloud/workflow_script/pull/22

As discussed @danxuliu  :) Don't hesitate to reach out for any questions, starter points, API stuff, also missing bits.

**Comment by blizzz** [comment]
@danxuliu had the wrong mention before

**Comment by nickvergessen** [comment]
Closing?

**Comment by blizzz** [comment]
We still might get a call has finished (and perhaps started) event in (just not for 18/this week). Even if with less information. Just flipping the milestone?

**Comment by marcelklehr** [comment]
An event for new chat messages would be nice, too.

**Comment by nickvergessen** [comment]
There are events available for other apps. A floe action for talk messages doesn't make a lot of sense to me.

What do you want to do with the event?

**Comment by marcelklehr** [comment]
I was thinking of automatically collecting links posted into a chat room in the bookmarks app, as an example. I've seen the existing events and they're great, but I think there are many instances like this one, where you don't want to impose that integration on all users, but rather have it optional via flow.

**Comment by marcelklehr** [comment]
In general I think exposing all events in flow would make a lot of sense as they exist already.

**Comment by nickvergessen** [comment]
Well there are a lot of events. Talk alone has around 20. And also not all of them should be accessible to all users, but e.g. moderators or no one. Also woth a flow about messages, you could create problems as there a flows which create messages and so on.

**Comment by marcelklehr** [comment]
I agree, maybe not all events should be available. Shooting yourself in the foot will inevitably be a problem with all apps that expose both operations and events. It might be worth thinking of a general solution for this.

**Comment by blizzz** [comment]
IIRC comments posted trigger an events via Dispatcher. @marcelklehr / Bookmarks could hook in there. I guess the question is mostly about configurability: whether to pick up links generally, or only from selected channels, perhaps giving pre-defined tags (not systemtags). With this in mind, flows might make sense, but it is not the only possible route to go.

**Comment by SamSpiri** [comment]
My use case:
I want to dump WhatsApp into trash and use NextCloud to share photos with my family. So when I put new family photo/video to NextCloud it should be posted into a Talk group so everybody could see and discuss.
